Chore/update cennzx

jordy25519 opened the pull request.

jordy25519 committed:
[WIP] fixing cennzx

jordy25519 committed:
CENNZX compiles

jordy25519 committed:
Format

jordy25519 committed:
Cargo deps fixed

jordy25519 committed:
Client compile and sync

jordy25519 committed:
Undo WIP sylo changes

jordy25519 committed:
Fmt

aliXsed commented on cli/src/cli.rs:
```diff
@@ -1,72 +1,38 @@
-// GNU General Public License for more details.
-
-// You should have received a copy of the GNU General Public License
-// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
```
Do we want to add our own header to this file?

aliXsed commented on cli/src/command.rs:
```diff
@@ -1,109 +1,156 @@
+		Ok(match id {
+			// "dev" => Box::new(chain_spec::dev::config()?),
+			// "azalea" => Box::new(chain_spec::azalea::config()?),
+			// "nikau" => Box::new(chain_spec::nikau::config()?),
```
Can we enable these now?

jordy25519 replied:
Yup I've got a different PR to enable these: https://github.com/cennznet/cennznet/pull/289

aliXsed commented on crml/cennzx/src/lib.rs:
```diff
@@ -66,13 +64,44 @@ pub struct LiquidityPrice<Balance> {
-	/// A function type to get an exchange address given the asset ID pair.
-	type ExchangeAddressGenerator: ExchangeAddressFor<Self::AssetId, Self::AccountId>;
-	type BalanceToUnsignedInt: From<<Self as pallet_generic_asset::Trait>::Balance> + Into<LowPrecisionUnsigned>;
-	type UnsignedIntToBalance: From<LowPrecisionUnsigned> + Into<<Self as pallet_generic_asset::Trait>::Balance>;
```
Great that we don't need these anymore. Well done.

aliXsed commented on crml/cennzx/src/lib.rs:
```diff
@@ -66,13 +64,44 @@ pub struct LiquidityPrice<Balance> {
+	/// Type for denoting asset balances
+	type Balance: Parameter + Member + AtLeast32BitUnsigned + Default + Copy;
+	/// Something which can provide asset management
+	type AssetSystem: SimpleAssetSystem<AccountId = Self::AccountId, Balance = Self::Balance, AssetId = Self::AssetId>;
```
Another nice improvement!

aliXsed reviewed (approved):
LVGTM!

jordy25519 committed:
Update CENNZX
Use multicurreny to simplify tests
Config CENNZX in runtime
Config CENNZX+GA rpcs

jordy25519 commented:
Had this open to show the diff w/ develop, closing now for #287
